### PR TITLE
fix emitter and make `Droppable` active when component is dragged

### PR DESCRIPTION
**Pull request by epoundor**

### CODE REVIEW

File: apps/frontend/src/components/Decorators/Draggable.tsx [hunk 1]
```diff
@@ -1,15 +1,13 @@
-import { useMitt } from '@/plugins/mitt/react-mitt';
-
 interface DraggableProps {
     children: React.ReactElement;
     type: string;
 }
 
 const Draggable: React.FunctionComponent<DraggableProps> = (props) => {
-    const emitter = useMitt();
+    const emitter = window.getEmitter();
```
Comment (YacineSteeve) [inline apps/frontend/src/components/Decorators/Draggable.tsx:7]: This MUST NOT be done.
Otherwise what's the utility of the `MittProvider` ?
The `emitter`'s getter in the `window` element should only be accessed inside `Preview` instances and subcomponents.

File: apps/frontend/src/App.tsx [hunk 2]
```diff
@@ -11,12 +11,12 @@ declare global {
 
 function App() {
     return (
-        <div>
-            <MittProvider>
+        <MittProvider>
+            <div>
```
Comment (YacineSteeve) [inline apps/frontend/src/App.tsx:15]: Not really matter, but OK.
Any particular reason to this ?

File: apps/frontend/src/components/Preview/PreviewDroppable/PreviewDroppable.tsx [hunk 3]
```diff
@@ -7,7 +7,15 @@ import PreviewDroppableStyle from './PreviewDroppable.module.css';
 const PreviewDroppable: React.FunctionComponent = () => {
     const dispatch = useDispatch();
     const [isHovered, setIsHovered] = useState<boolean>(false);
-
+    const [isDraggabled, setIsDraggisDraggabled] = useState<boolean>(false);
```
Comment (YacineSteeve) [inline apps/frontend/src/components/Preview/PreviewDroppable/PreviewDroppable.tsx]: Check your variable names before PRs 😂 
Should be replaced by `isDraggedOver` and `setIsDraggedOver`.

File: apps/frontend/src/components/Preview/PreviewDroppable/PreviewDroppable.tsx [hunk 4]
```diff
@@ -7,7 +7,15 @@ import PreviewDroppableStyle from './PreviewDroppable.module.css';
...
+    const emitter = window.parent.getEmitter();
```
Comment (YacineSteeve) [inline apps/frontend/src/components/Preview/PreviewDroppable/PreviewDroppable.tsx:11]: Nope

File: apps/frontend/src/components/Preview/PreviewDroppable/PreviewDroppable.tsx [hunk 5]
```diff
@@ -7,7 +7,15 @@ import PreviewDroppableStyle from './PreviewDroppable.module.css';
 const PreviewDroppable: React.FunctionComponent = () => {
     const dispatch = useDispatch();
     const [isHovered, setIsHovered] = useState<boolean>(false);
-
+    const [isDraggabled, setIsDraggabled] = useState<boolean>(false);
```
Comment (YacineSteeve) [inline apps/frontend/src/components/Preview/PreviewDroppable/PreviewDroppable.tsx:10]: Should be `isDraggedOver` and `setIsDraggedOver`

File: apps/frontend/src/components/PreviewComponentWrapper/PreviewComponentWrapper.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
+// import { useMitt } from '@/plugins/mitt/react-mitt';
+import { mitt } from '@/plugins/mitt';
+import { addComponent } from '@/store/previewTree/previewTreeSlice';
+import { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import Droppable from '../Decorators/Droppable';
+import Icon from '../Decorators/Icon';
+import PreviewComponentWrapperStyle from './PreviewComponentWrapper.module.css';
+
+interface PreviewComponentWrapperProps {
```
Comment (YacineSteeve) [inline apps/frontend/src/components/PreviewComponentWrapper/PreviewComponentWrapper.tsx:10]: Why did you move this component from the `components/Preview/` directory ?

File: apps/frontend/src/plugins/mitt/index.ts [hunk 7]
```diff
@@ -111,3 +111,6 @@ export function mitt<Events extends Record<EventType, unknown>>(
         }
     };
 }
+
+const emitter = mitt();
```
Comment (YacineSteeve) [inline apps/frontend/src/plugins/mitt/index.ts:115]: The `emitter` is then instanciated twice. Here and in `react-mitt`.
Must keep it as it was.

File: apps/frontend/src/plugins/mitt/react-mitt.tsx [hunk 8]
```diff
@@ -11,4 +11,6 @@ export const MittProvider: React.FunctionComponent<React.PropsWithChildren> = ({
     return <MittContext.Provider value={emitter}>{children}</MittContext.Provider>;
 };
 
-export const useMitt = (): MittContextType => useContext(MittContext);
+export const useMitt = (): MittContextType => {
+    return useContext(MittContext);
```
Comment (YacineSteeve) [inline apps/frontend/src/plugins/mitt/react-mitt.tsx:15]: Any particular reason ?

**Review by YacineSteeve — CHANGES_REQUESTED** [review]

File: apps/frontend/src/components/Decorators/Draggable.tsx [hunk 9]
```diff
@@ -1,18 +1,16 @@
...
```
Comment (eliotfgn) [inline apps/frontend/src/components/Decorators/Draggable.tsx:7]: I think the hook can be keeped

**Review by eliotfgn — CHANGES_REQUESTED** [review]
The useMitt can be preserved !!!

**Review by YacineSteeve — APPROVED** [review]
Much better, I think

**Review by eliotfgn — APPROVED** [review]
